frontend/public/components/modals/remove-volume-modal.tsx

```diff
@@ -0,0 +1,90 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory';
+import { history, resourcePathFromModel } from '../utils';
+import { getVolumeType, K8sKind, k8sPatch, K8sResourceKind } from '../../module/k8s/';
+import { RowVolumeData } from '../volumes-table';
+
+export const RemoveVolumeModal: React.FC<RemoveVolumeModalProps> = (props) => {
+  const [inProgress, setInProgress] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState('');
+
+  const getRemoveVolumePatch = (resource, rowVolumeData: RowVolumeData) => {
+    const location = resource.kind === 'Pod' ? 'spec' : 'spec.template.spec';
+    const dirLocation = resource.kind === 'Pod' ? '/spec' : '/spec/template/spec';
```

Comment:
We should be able to remove all the special casing here and just use hard-coded paths since we can't support remove for pods.